### PR TITLE
CNDE-1602: SAS to SQL RTR: Migrate CASE_COUNT SAS dimension to SQL

### PR DESCRIPTION
**Pull request by johneipe**
## Notes

SAS to SQL RTR: Migrating CASE_COUNT SAS dimension to SQL procedures

## JIRA

- **Related story**: [CNDE-1602](https://cdc-nbs.atlassian.net/browse/CNDE-1602)

## Checklist

- [x] PR focuses on a single story.
- [ ] New unit tests added and ensured they pass.
- [x] Service has been tested in local and it works as expected.
- [ ] Documentation has been updated for this code change (if needed).
- [ ] Code follows the Java Coding Conventions (https://www.oracle.com/java/technologies/javase/codeconventions-programmingpractices.html).

## Types of changes

What types of changes does this PR introduces?

- [ ] Bugfix
- [x] New feature
- [ ] Breaking change

## Testing

- [ ] Does this PR has >90% code coverage?
- [ ] Is the screenshot attached for code coverage?
- [ ] Does the `gradle build` pass in your local? 
- [ ] Is the `gradle build` logs attached?

### CODE REVIEW

File: investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/repository/model/dto/InvestigationTransformed.java [hunk 1]
```diff
@@ -14,6 +14,9 @@ public class InvestigationTransformed {
     private String legacyCaseId;
     private Long phcInvFormId;
     private String rdbTableNameList;
+    private Long investigationCount;
+    private Long caseCount;
```
Comment (upasanapattnaik-eq) [inline investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/repository/model/dto/InvestigationTransformed.java:18]: 👍

File: liquibase-service/src/main/resources/db/rdb_modern/routines/021-sp_nrt_case_count_postprocessing-001.sql [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+    src.adt_hsptl_key,
+    src.inv_start_dt_key,
+    src.diagnosis_dt_key,
+    src.inv_rpt_dt_key,
+    src.geocoding_location_key
+from
+    #CASE_COUNT src
+left outer join dbo.CASE_COUNT tgt
+    on src.investigation_key = tgt.investigation_key
+where tgt.investigation_key is null
```
Comment (upasanapattnaik-eq) [inline liquibase-service/src/main/resources/db/rdb_modern/routines/021-sp_nrt_case_count_postprocessing-001.sql:168]: 👍

File: liquibase-service/src/main/resources/db/odse/routines/010-sp_investigation_event-002.sql [hunk 3]
```diff
@@ -610,7 +612,46 @@ BEGIN
                                                )
                                            FOR json path,INCLUDE_NULL_VALUES
                                        ) AS notification_history
-                                  ) AS notification_history
+                                  ) AS notification_history,
+                                  (select
+                                        (
+	                                        select
+		                                        phcase.group_case_cnt                                           as investigation_count,
+		                                        round(COALESCE (gcs.group_case_cnt, phcase.group_case_cnt),0)   as case_count,
```
Comment (upasanapattnaik-eq) [inline liquibase-service/src/main/resources/db/odse/routines/010-sp_investigation_event-002.sql:620]: So we pick the value the count that exists between SummaryFormQ or the PHC's group_case_count at the core of the logic?

**Review by upasanapattnaik-eq — COMMENTED** [review]
Will the unit test be taken up in a separate ticket?

**Review by sveselev — CHANGES_REQUESTED** [review]
Failed unit tests should be fixed (otherwise deployment is not possible)

File: liquibase-service/src/main/resources/db/odse/routines/010-sp_investigation_event-002.sql (hunk 3 above)
Comment (johneipe) [reply liquibase-service/src/main/resources/db/odse/routines/010-sp_investigation_event-002.sql:620]: yes, that is correct.

File: liquibase-service/src/main/resources/db/changelog/db.rdb_modern.changelog-16.1.yaml [hunk 4]
```diff
@@ -447,3 +447,17 @@ databaseChangeLog:
...
+      id: 60
+      author: liquibase
+      changes:
+        - sqlFile:
+            path: 020-create_nrt_investigation-003.sql
+            splitStatements: false
+  - changeSet:
+      id: 61
+      author: liquibase
+      changes:
```
Comment (sveselev) [inline liquibase-service/src/main/resources/db/changelog/db.rdb_modern.changelog-16.1.yaml:460]: please add
```
      runOnChange: true
```
here

**Review by sveselev — CHANGES_REQUESTED** [review]

File: liquibase-service/src/main/resources/db/changelog/db.rdb_modern.changelog-16.1.yaml [hunk 5]
```diff
@@ -447,3 +447,17 @@ databaseChangeLog:
         - sqlFile:
             path: 007-create_nrt_investigation-002.sql
             splitStatements: false
+  - changeSet:
+      id: 60
+      author: liquibase
```
Comment (sveselev) [inline liquibase-service/src/main/resources/db/changelog/db.rdb_modern.changelog-16.1.yaml:452]: please add
```      runOnChange: true```
here

**Review by sveselev — APPROVED** [review]

**Pull request merged**